jenkins-job-components/src/main/java/com/ofg/job/component/JobComponent.java

```diff
@@ -0,0 +1,20 @@
+package com.ofg.job.component;
+
+import javaposse.jobdsl.dsl.Job;
+
+/**
+ * Interface for configurations applicable on Job.
+ */
+public interface JobComponent<J extends Job> {
+
+    JobComponent<? extends Job> EMPTY = job -> {};
+    
+    default <NJ extends J> JobComponent<NJ> and(JobComponent<NJ> other) {
```

Comment:
hm wouldn't it be possible with such jobs to do:
```
AbstractJobConfigurer() >>
AbstractJobConfigurer() >>
MultiJobConfigurer() >>
MultiJobConfigurer() >>>
AbstractJobConfigurer() >>
MavenJobConfigurer()
```?
If that's not the case, then please prove me wrong and propose such methods with tests, since I've no idea how to define it in a way that wouldn't allow this to happen.